src/onelogin/saml2/idp_metadata_parser.py

```diff
@@ -108,9 +126,12 @@ def parse(idp_metadata):
 
                     if idp_entity_id is not None:
                         data['idp']['entityId'] = idp_entity_id
+
                     if idp_sso_url is not None:
                         data['idp']['singleSignOnService'] = {}
                         data['idp']['singleSignOnService']['url'] = idp_sso_url
+                        data['idp']['singleSignOnService']['binding'] = preferred_sso_binding
```

Comment:
I will rename `preferred_sso_binding` to `required_sso_binding`, and actually make it behave like so. I think this will be less ambiguous and therefore more developer-friendly than `preferred`.